portfolio/src/main/webapp/script.js

```diff
@@ -17,12 +17,45 @@
  */
 function addRandomFact() {
   const facts =
-      ['I have a pet dog', 'I like to code'];
+      ['I have a pet dog!', 'I like to code!', 'Marquees are my favourite HTML element!'];
 
   // Pick a random fact.
   const fact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
-}
\ No newline at end of file
+}
+
+/**
+ * Controls slideshow image display on the page
+ */
+var slideIndices = [0, 0, 0]; // array of slide indices for each slide deck
+for(var i = 0; i < slideIndices.length; i++) {
+    showSlides(i, slideIndices[i]);
+}
+
+function nextSlide(deckNum, direction) {
+  showSlides(deckNum, slideIndices[deckNum] += direction);
+}
+
+// Returns corresponding class name for given deckNum
+function deckNumToName(deckNum) {
```

Comment:
Its ok to use strings instead of numbers. So need to have this method.

Also, If I ever see in the wild nextSlide(1, 2), it is hard to tell what this function does without looking at its definition. It would be much easier to read if I saw nextSlide(PLACES_DECK, NEXT_SLIDE) where PLACES_DECK and NEXT_SLIDE are constants defined at the top of the file. PLACES_DECK could be defined as the string "dog-slides" and NEXT_SLIDE could be the integer 1 (since you are making this work using arithmetic). For the string constant, you might wonder why not just hard code the string. This is just convention that if you have the same string multiple times in the file might as well store it in a constant cause then it would be a lot easier to modify (if it ever needs to be) since you would only have to change it in one place instead of in multiple places in the file.

If something doesn't make sense, just add to this comment with questions.

Reply:
This could be a good case to use an "enum".

JS doesn't have an enum keyword, but the style guide has a commonly-used way to work around that. You can ignore the part talking about @enum annotations. https://google.github.io/styleguide/jsguide.html#features-objects-enums